src/net/raw/devices/devices-macos.c

```diff
@@ -0,0 +1,102 @@
+/*
+ * Copyright 2015 click2stream, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <stdio.h>
+#include <string.h>
+#include <stdlib.h>
+#include <unistd.h>
+#include <sys/ioctl.h>
+#include <net/if.h>
+#include <net/if_arp.h>
+#include <net/if_dl.h>
+#include <netinet/in.h>
+#include <ifaddrs.h>
+
+#include "devices.h"
+#include "devices-utils.h"
+#include "utils.h"
+
+static int get_mac_address(const char* dname, unsigned char* buffer) {
+    struct ifaddrs* iflist;
+    struct ifaddrs* cur;
+
+    if (getifaddrs(&iflist) == 0) {
+        for (cur = iflist; cur; cur = cur->ifa_next) {
+            if ((cur->ifa_addr->sa_family == AF_LINK) &&
+                    (strcmp(cur->ifa_name, dname) == 0) &&
+                    cur->ifa_addr) {
```

Comment:
You're testing that `cur->ifa_addr` isn't NULL after it was dereferenced on line 38. Shouldn't the condition be this way:
```
if (cur->ifa_addr
        && cur->ifa_addr->sa_family == AF_LINK
        && strcmp(cur->ifa_name, dname) == 0) { ... }
```
?

Reply:
argh, just some leftover copy-paste, thanks for noticing